src/main/java/org/repodriller/domain/ChangeSet.java

```diff
@@ -16,45 +16,148 @@
 
 package org.repodriller.domain;
 
-import java.text.SimpleDateFormat;
 import java.util.Calendar;
+import java.util.Set;
+import java.util.TreeSet;
 
 /**
- * A ChangeSet is metadata that uniquely identifies a Commit from an SCM.
- *
- * @author Mauricio Aniche
+ * A ChangeSet is a POJO for the metadata about a Commit from an SCM.
+ * It's "everything but the diff".
  */
 public class ChangeSet {
 
-	private final Calendar date;
-	private final String id;
+	public enum Contributor {
+	    AUTHOR, COMMITTER
+	};
 
-	public ChangeSet(String id, Calendar date) {
+	private String id; /* Unique within an SCM. Example: the commit hash. */
+	private String message;
+
+	private CommitContributor author; /* Person who committed it. */
+	private CommitContributor committer; /* Person who merged it. This is probably the field you want. */
+
+	private Set<String> parentIds; /* Parent(s) of this commit. Only "merge commits" have more than one parent. Does the root commit have 0? */
+
+	private Set<String> branches; /* Set of branches that contain this commit, i.e. the branches for which this commit is an ancestor of the most recent commit in that branch. */
```

Comment:
Yeah, this will definitely increase the memory footprint. Maybe we should indeed discard `ChangeSet`s after they are used.

Reply:
Discarding `ChangeSet`s after they are used will definitely reduce the time of the maximum footprint (decreasing `ChangeSet` footprint as the expected footprint from `CommitVisitors` rises). I'll look into adding this, should just be a few lines of code.

Reply:
See 1b7affc.